sdk/javascript/v1/client.js

```diff
@@ -264,6 +272,124 @@ class Client {
             call.end();
         });
     }
+
+    sign(req, oidc_client_id = "sigstore") {
+        const provider = req.u[1][1];
+        let key_file_found = false;
+
+        // INFO: Assume that the request is key based if the provider array
+        // first item is a buffer (key file)
+        try {
+            key_file_found = Buffer.isBuffer(provider[0]);
+        } catch (error) {
+            if (!error.message.includes("Cannot read properties of undefined")) {
+                throw err;
+            }
+        }
```

Comment:
is there no explicit field that is set for key-based and oidc-based signing? afaict `req` is the `SignRequest` type, so we can decide without assumptions. is this the case or not?

Reply:
```js
let oidc_provider = req.u[1][0];
let key_provider = req.u[1][1];
```

can we use the check if key_provider/oidc_signing is not null checks